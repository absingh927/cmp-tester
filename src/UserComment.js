import React from "react";
import styled from "styled-components";

const Container = styled.div`
  padding: 1.25rem 0.5rem;
  border-top: 2px solid rgba(186, 186, 186, 0.2);
  border-bottom: 2px solid rgba(186, 186, 186, 0.2);
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
`;

const ContainerLeft = styled.div`
  img {
    border-radius: 50%;
    max-width: 40px;
  }
`;

const ContainerRight = styled.div`
  margin-left: 1rem;
`;

const Title = styled.div`
  color: #000;
  font-weight: 800;
  display: flex;
  align-items: center;
`;

const Pill = styled.div`
  background-color: #000;
  color: #fff;
  text-transform: uppercase;
  padding: 0.2rem 0.625rem;
  border-radius: 10rem;
  margin: 0 0.5rem;
  font-size: 0.625rem;
  font-weight: 600;
`;

const Duration = styled.div`
  color: #bababa;
  text-transform: uppercase;
  font-size: 0.75rem;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
`;

const ActionBtn = styled.button`
  background-color: transparent;
  color: #bababa;
  margin: 0 0.25rem;
  border: none;
  font-seize: 0.75rem;
  cursor: pointer;

  $:focus {
    outline: none;
  }

  span {
    color: #919191;
  }
`;

const Body = styled.div`
  margin: 0.5rem 0;
  line-height: 1.5rem;
`;

const UserComment = () => {
  return (
    <Container>
      <ContainerLeft>
        <img src="http://placekitten.com/100/100" alt="user" />
      </ContainerLeft>
      <ContainerRight>
        <Title>
          Brad
          <Pill>Author</Pill>
          <Duration>&#183; 2 minutes ago</Duration>
        </Title>
        <Body>
          So what the German automaker is likely to focus on today is the bigger
          picture. This will be the first time we see the Taycan free from any
          prototype bodywork
        </Body>
        <Actions>
          <ActionBtn style={{ paddingLeft: 0, marginLeft: 0 }}>REPLY</ActionBtn>
          <ActionBtn>
            <span>21</span>&nbsp;REPLIES
          </ActionBtn>
          <ActionBtn>123</ActionBtn>
          <ActionBtn>0</ActionBtn>
        </Actions>
      </ContainerRight>
    </Container>
  );
};

export default UserComment;
