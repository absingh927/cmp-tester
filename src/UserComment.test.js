import React from "react";
import { shallow } from "enzyme";

import UserComment from "./UserComment";

describe("UserComment", () => {
  it("should render correctly with sample data", () => {
    const sampleData = {
      name: "Name",
      role: "tester",
      duration: "20",
      comment: "sample small component",
      replies: 10,
      upVote: 100,
      downVote: 50,
    };

    const cmp = shallow(<UserComment data={sampleData} />);
    expect(cmp).toMatchSnapshot();
  });
