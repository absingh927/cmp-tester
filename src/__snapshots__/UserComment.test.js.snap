// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComment should render correctly with no props 1`] = `
<styled.div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <styled.div>
    <img
      alt="user"
      src="http://placekitten.com/100/100"
    />
  </styled.div>
  <styled.div>
    <styled.div>
      Name
      <styled.div>
        tester
      </styled.div>
      <styled.div>
        · 
        20
         minutes ago
      </styled.div>
    </styled.div>
    <styled.div>
      sample small component
    </styled.div>
    <styled.div>
      <styled.button
        isHovering={false}
        style={
          Object {
            "marginLeft": 0,
            "paddingLeft": 0,
          }
        }
      >
        REPLY
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <span>
          10
        </span>
         REPLIES
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-up"
        />
         
        100
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-down"
        />
         
        50
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`UserComment should render correctly with props 1`] = `
<styled.div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <styled.div>
    <img
      alt="user"
      src="http://placekitten.com/100/100"
    />
  </styled.div>
  <styled.div>
    <styled.div>
      Name
      <styled.div>
        tester
      </styled.div>
      <styled.div>
        · 
        20
         minutes ago
      </styled.div>
    </styled.div>
    <styled.div>
      sample small component
    </styled.div>
    <styled.div>
      <styled.button
        isHovering={false}
        style={
          Object {
            "marginLeft": 0,
            "paddingLeft": 0,
          }
        }
      >
        REPLY
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <span>
          10
        </span>
         REPLIES
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-up"
        />
         
        100
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-down"
        />
         
        50
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`UserComment should render correctly with sample data 1`] = `
<styled.div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <styled.div>
    <img
      alt="user"
      src="http://placekitten.com/100/100"
    />
  </styled.div>
  <styled.div>
    <styled.div>
      Name
      <styled.div>
        tester
      </styled.div>
      <styled.div>
        · 
        20
         minutes ago
      </styled.div>
    </styled.div>
    <styled.div>
      sample small component
    </styled.div>
    <styled.div>
      <styled.button
        isHovering={false}
        style={
          Object {
            "marginLeft": 0,
            "paddingLeft": 0,
          }
        }
      >
        REPLY
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <span>
          10
        </span>
         REPLIES
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-up"
        />
         
        100
      </styled.button>
      <styled.button
        isHovering={false}
      >
        <i
          className="fas fa-chevron-down"
        />
         
        50
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>
`;
